src/main/java/org/mockito/internal/verification/checkers/AtLeastXNumberOfInvocationsChecker.java

```diff
@@ -28,7 +29,17 @@ public void check(List<Invocation> invocations, InvocationMatcher wanted, int wa
             Location lastLocation = finder.getLastLocation(actualInvocations);
             reporter.tooLittleActualInvocations(new AtLeastDiscrepancy(wantedCount, actualCount), wanted, lastLocation);        
         }
-        
+
+        removeAlreadyVerified(actualInvocations);
         invocationMarker.markVerified(actualInvocations, wanted);
     }
+
+    private void removeAlreadyVerified(List<Invocation> invocations) {
```

Comment:
Since this method is duplicate from the method added in the class below, maybe we can extract an superclass `InvocationsChecker`. This class abstracts away all the logic regarding obtaining and reporting the failures, the constructor as well as this new method.